helm/app-operator-chart/templates/rbac.yaml

```diff
@@ -0,0 +1,110 @@
+apiVersion: rbac.authorization.k8s.io/v1beta1
+kind: ClusterRole
+metadata:
+  name: {{ .Values.name }}
+  labels:
+    app: {{ .Values.name }}
+    giantswarm.io/service-type: "managed"
+rules:
+- apiGroups:
+  - apiextensions.k8s.io
+  resources:
+  - customresourcedefinitions
+  verbs:
+  - "*"
+- apiGroups:
+    - application.giantswarm.io
+  resources:
+    - apps
+  verbs:
+    - get
+    - list
+    - patch
+    - update
+    - watch
+- apiGroups:
+    - core.giantswarm.io
+  resources:
+    - charts/status
+  verbs:
+    - create
+    - patch
+    - update
+- apiGroups:
+  - core.giantswarm.io
+  resources:
+  - charts
+  verbs:
+  - get
+  - list
+  - patch
+  - update
+  - watch
+- apiGroups:
+  - core.giantswarm.io
+  resources:
+  - charts/status
+  verbs:
+  - create
+  - patch
+  - update
```

Comment:
chart/status is set by chart-operator and should not be changed by app-operator, it should only be able to get the status (and maybe list?) right?

also this and above here would be only in case it creates chart CRs in control plane (which we want in future), but we need the same in tenant-clusters too

Reply:
btw, maybe we add a note/comment here in the YAML that this is only for creating chart CRs in control plane (in the future) and not for tenant clusters as those RBAC roles are managed inside the tenants

Reply:
as per https://github.com/giantswarm/app-operator/pull/7#discussion_r245029875 we do not need any chart CR related rights in the role as they should be bound to the kubeconfig it gets not to the operator itself

Reply:
@puja108 But don't `app-operator` need to have creation permission for `charts` CR at least?

Reply:
it will get it through `kubeconfig` for the cluster it needs to, not from this RBAC role